air/src/execution_step/boxed_value/stream.rs

```diff
@@ -38,17 +39,20 @@ pub struct Stream {
 }
 
 impl Stream {
-    pub(crate) fn from_generations_count(previous_count: usize, current_count: usize) -> Self {
-        let last_generation_count = 1;
+    pub(crate) fn from_generations_count(
+        previous_count: impl Into<GenerationIdx>,
```

Comment:
`Into<ANewtypeWrapper>` shouldn't be used in the library code as it reduces type safety (making the newtype useless).  However, it can be used in test library, as the tests are written manually.

Reply:
Agree, but I'll delete this function literally in the next PR.

Reply:
fixed for consistency